File: algorithm/src/main/java/baekjoon/p04564/Main.java
package baekjoon.p04564;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (true) {
            int input = sc.nextInt();
            if (input == 0) break;
            for (int solve : solve(input)) {
                System.out.print(solve + " ");
            }
            System.out.println();
        }
    }

    private static List<Integer> solve(int input) {
        int n = input;
        List<Integer> ret = new ArrayList<>();

        while (n >= 10) {
            ret.add(n);
            n = mul(n);
        }
        ret.add(n);

        return ret;
    }

    private static int mul(int n) {
        int ret = 1;

        while (n != 0) {
            ret *= (n % 10);
            n /= 10;
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p04564/Main.kt
package baekjoon.p04564

fun main(args: Array<String>) {
    while (true) {
        val input = readLine()!!.toInt()
        if (input == 0) break
        println(solve(input).joinToString(" "))
    }
}

private fun solve(input: Int): List<Int> {
    val ret = ArrayList<Int>()
    var n = input

    while (n >= 10) {
        ret.add(n)
        n = mul(n)
    }
    ret.add(n)

    return ret
}

private fun mul(n: Int): Int {
    var ret = 1
    var nn = n

    while (nn != 0) {
        ret *= (nn % 10)
        nn /= 10
    }

    return ret
}